ui/dashboard/src/pages/signin/signin.tsx

```diff
@@ -0,0 +1,79 @@
+import { useNavigate } from 'react-router-dom';
+import { authenticationUrl } from '@api/auth';
+import { useAuth } from 'auth';
+import { urls } from 'configs';
+import { PAGE_PATH_AUTH_SIGNIN } from 'constants/routing';
+import { setCookieState } from 'cookie';
+import { useSubmit } from 'hooks';
+import { IconEmail, IconGoogle } from '@icons';
+import { Button } from 'components/button';
+import Icon from 'components/icon';
+import AuthWrapper from './elements/auth-wrapper';
+
+const SignIn = () => {
+  const navigate = useNavigate();
+  const { isGoogleAuthError, setIsGoogleAuthError } = useAuth();
+
+  const { onSubmit: onGoogleLoginHandler, submitting } = useSubmit(() => {
+    const state = `${Date.now()}`;
+    setCookieState(state);
+    setIsGoogleAuthError(false);
+
+    return authenticationUrl({
+      state,
+      redirectUrl: urls.AUTH_REDIRECT,
+      type: 2 // Google auth type
+    }).then(response => {
+      if (response.url) {
+        window.location.href = response.url;
+      }
+    });
+  });
+
+  return (
+    <AuthWrapper>
+      <div className="grid gap-6">
+        <h1 className="text-gray-900 typo-head-bold-huge">
+          {`Sign in to Bucketeer`}
+        </h1>
+        {isGoogleAuthError && (
+          <p className="text-accent-red-500 typo-para-medium">
+            {` No account found for this Google Account. Please check the email
+          entered or try a different Google Account.`}
+          </p>
+        )}
+      </div>
+      <div className="flex flex-col gap-4 mt-10">
+        <Button
+          variant="secondary-2"
+          onClick={() => {
+            navigate(PAGE_PATH_AUTH_SIGNIN);
+            setIsGoogleAuthError(false);
+          }}
+        >
+          <Icon icon={IconEmail} />
+          {`Sign in With Email`}
+        </Button>
```

Comment:
This button must only show when the `DEMO_SIGN_IN_ENABLED`.

Reference.
https://github.com/bucketeer-io/bucketeer/blob/main/ui/web-v2/src/pages/auth/signin.tsx#L30-L37

Also, please check `DEMO_SIGN_IN_ENABLED` in the root http://localhost:8000/auth/signin if the user tries to access it directly. We should redirect to the login page if the user is not logged in or the 404 page if it is logged in.